src/gridcoin/gridcoin.cpp

```diff
@@ -28,6 +28,68 @@ void Scraper(bool bSingleShot = false);
 void ScraperSubscriber();
 
 namespace {
+//!
+//! \brief Display a message that warns about a corrupted blockchain database.
+//!
+//! For the GUI, this displays a modal dialog. It prints a message to the log
+//! and to stderr on headless nodes.
+//!
+void ShowChainCorruptedMessage()
+{
+    uiInterface.ThreadSafeMessageBox(
+        _("WARNING: Blockchain data may be corrupted.\n\n"
+            "Gridcoin detected bad index entries. This may occur because of an "
+            "unexpected exit, a power failure, or a late software upgrade.\n\n"
+            "Please exit Gridcoin, open the data directory, and delete:\n"
+            " - the blk****.dat files\n"
+            " - the txleveldb folder\n\n"
+            "Your wallet will re-download the blockchain. Your balance may "
```

Comment:
Might be a good idea to add to this message to mention that you also need to delete the accrual folder

Reply:
The wallet will delete it automatically if there is no chain data when it starts.

Reply:
Yes. That is true. I have asked @iFoggz to extend the snapshot download functionality to allow to delete the blockchain data and sync from zero instead of the snapshot. This is simply a subset of the snapshot download functionality. Quite frankly in the case where there is already a checkpoint mismatch on startup, we should not make the user delete those files themselves and instead warn them that the blockchain data is corrupt and will be deleted and then offer them the choice of sync from zero (preferably) or snapshot.

Reply:
Yep, ideally the dialog should offer those options to the user. I started with a simple message until that framework is ready.

Reply:
Let's go with this and we can build on it.

Reply:
Note that with the additional checkpoint in #2087, this should catch many of the newbie-bug forked nodes as they upgrade.